pkg/apis/networking/v1alpha1/certificate_lifecycle.go

```diff
@@ -58,6 +58,14 @@ func (c *Certificate) IsReady() bool {
 		cs.GetCondition(CertificateConditionReady).IsTrue()
 }
 
+// IsFailed returns true is the Certificate ready is false
+// and the Certificate resource has been observed.
+func (c *Certificate) IsFailed() bool {
```

Comment:
Do we need that extra method? Could the user not just use `!IsReady`?

Reply:
@ReToCode Thanks for your reply. 
I think `IdReady` is not enough because the `ConditionStatus` has three possible value: `true`, `false` or `unknown`. `IsReady` only checks whether it's `true`. And the new `IsFailed` check whether it's `false`.